### ISSUE TITLE
Rewrite the core in something different than bash

### ISSUE THREAD

**Issue by Doeme** [IBT-FMI/gebuilder#32]
This is not strictly necessary, but having the core written in a programming language that has access to more linux/glibc-features might be beneficial for dropping privileges when operating on a build directory (e.g. seccomp, namespaces, etc..)

The core can execute a bash-process, read the right files and pipe them into the bash-stdin.
Also, we might benefit from a less botched log-redirection feature.

I'm open for suggestions. C is of course the most feature-rich, python is usually installed on gentoo system so it might be worth a shot, and personally I'm a fan of DLang, but this has the drawback of depending on a not widely distributed compiler.

**Comment by TheChymera** [comment]
Sounds good, I'm only wondering if this is not a bit premature, after all we're still in the phase of stabilizing a prototype and determining its core features. If you do want to go for this, you'd have to consider how much time you are able to put into it, and if that is “not a lot” it would need to be a language in which I can also contribute, which from the ones you mentioned is only Python.

**Comment by Doeme** [comment]
Yes, this is definately premature. It's just something that I think can be improved at some point.

I'll add the issue to a far away milestone, s.t. this is clear.
